### PR TITLE
local_docker: make it more explicit when a test is killed when reaching a timeout

### PR DESCRIPTION
**Pull request by laurentsenta**
Relates to #1542
Closes https://github.com/testground/testground/issues/1551

When a job reaches the timeout, it's not very clear that it was canceled.

### CODE REVIEW

**Comment by galargh** [comment]
Is it necessarily what we want? To me a timeout feels closer to a failure rather than cancellation because I expect the latter to be intentional and initiated by user. I'd see reaching a timeout as a failure with timeout as a reason. Do you know other testing frameworks that make distinction between failure and cancellation? Maybe it's just my initial thinking that's off :P

**Comment by laurentsenta** [comment]
I see your point, "canceled" might sound like a user action.

I would prefer being explicit:
- a failure happens when your code failed,
- Testground might end a test for various reasons, because it's configured as such, timeout is one,

To answer directly your question: I don't think go does, but it's famously know to be the bare minimum, jest logs a "timeout reached" I think, I'd have to look into others.

What about github actions? There is a global timeout on the platform, do you know what happens when we reach this timeout?

**Comment by galargh** [comment]
Here's how it goes for GH Actions:
- timeout case: https://github.com/galargh/.github/actions/runs/3704316375
- cancelled case: https://github.com/galargh/.github/actions/runs/3704333550

In both cases the job is marked as cancelled. but in the timeout case the workflow run is marked as failure whereas in the cancelled case it's marked as cancelled.

**Comment by laurentsenta** [comment]
Ok, let's mimick github on that one, please don't merge this PR until I have time to rework it, I'd like to at least log an explicit message on timeout.

**Comment by laurentsenta** [comment]
@galargh PR updated and ready for review!

File: pkg/runner/local_docker.go
Comment (galargh) [inline pkg/runner/local_docker.go:298]:
```suggestion
			result.Outcome = task.OutcomeCanceled
```

**Review by galargh — APPROVED** [review]
We lost `OutcomeCancelled` in the process :P Looks good otherwise. I'll let you decide if we need a test case for this.

Approving so that you can merge after applying suggestions 👍 👍 👍

**Pull request merged**
